IdentityCore/src/throttling/thumbprint_calculation/MSIDSilentRequestThumbprintCalculator.m

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDSilentRequestThumbprintCalculator.h"
+#import "MSIDThumbprintWrapperObject.h"
+#import "MSIDOAuth2Constants.h"
+
+@interface MSIDSilentRequestThumbprintCalculator ()
+
+@property (nonatomic) NSMutableDictionary *requestParameters;
+@property (nonatomic) NSSet *strictThumbprintIncludeSet; //white list for items to include for strict request thumbprint calculation
+@property (nonatomic) NSSet *fullThumbprintExcludeSet; //black list for items to exclude from full request thumbprint calculation
+
+@end
+
+@implementation MSIDSilentRequestThumbprintCalculator
+
+- (instancetype)initWithParamaters:(NSDictionary *)parameters
+                       endpointUrl:(NSString *)endpointUrl
+                             realm:(NSString *)realm
+                       environment:(NSString *)environment
+                     homeAccountId:(NSString *)homeAccountId
+{
+    self = [super init];
+    if (self)
+    {
+        _requestParameters = [parameters mutableCopy];
```

Comment:
I'm not sure we always have these values not nil. Please check nil before adding to the calculate params list